compiler/luci/profile/src/CircleNodeOrigin.cpp

```diff
@@ -141,13 +141,22 @@ namespace luci
 
 bool has_origin(const luci::CircleNode *circle_node)
 {
-  return circle_node->annot<CircleNodeOriginAnnotation>() != nullptr;
+  if (circle_node->annot<CircleNodeOriginAnnotation>() == nullptr)
+    return false;
+  if (circle_node->annot<CircleNodeOriginAnnotation>()->origin() == nullptr)
+    return false;
+  return true;
 }
 
 void add_origin(luci::CircleNode *circle_node, const std::shared_ptr<CircleNodeOrigin> origin)
 {
+  if (origin == nullptr)
+    return;
+
+  auto new_origin = composite_origin({get_origin(circle_node), origin});
```

Comment:
This is not the way you said in the comments. Is it a former?

Reply:
This draft is for discussing and recording where to fix. So the code is not fixed :)